internal/pkg/scheduler/scheduler.go

```diff
@@ -78,6 +87,9 @@ func New(schedules []Schedule, opts ...OptFunc) (*Scheduler, error) {
 	return s, nil
 }
 
+// Run executes all scheduled function according to their schedules.
+// Schedule Interval times are garunteed minium values (if a execution takes a very long time, the scheduler will wait Interval before running the function again).
```

Comment:
garunteed -> guaranteed

"if a execution takes a very long time" -> "if the execution takes a very long time"